modules/serverless-rn-1-4-0.adoc

```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+// * serverless/release-notes.adoc
+
+[id="serverless-rn-1-4-0_{context}"]
+= Release Notes for Red Hat {ServerlessProductName} Technology Preview 1.4.0
+
+[id="new-features-1-4-0_{context}"]
+== New features
+* {ServerlessProductName} 1.4.0 is available on {product-title} 4.2 and newer versions.
+* {ServerlessProductName} has been updated to use Knative Serving 0.11.1.
+* {ServerlessProductName} has been updated to use Knative Client (`kn` CLI) 0.11.0.
+* {ServerlessProductName} has been updated to use Knative Serving Operator 0.11.1.
+* Knative Client is now available for download through the *Command Line Tools* page in the {product-title} web console.
+* The `KnativeServing` object's API group has changed in this release from `serving.knative.dev` to `operator.knative.dev`. This migration is completed automatically when {ServerlessOperatorName} 1.4.0 is installed.
++
+[IMPORTANT]
+====
+You will need to adjust all of your scripts to reflect the new API group. The {ServerlessProductName} installation instructions have been updated to provide details on how to complete this adjustment.
```

Comment:
```suggestion
You will need to adjust all of your scripts to reflect the new API group. The {ServerlessProductName} installation instructions have been updated to the new API group as well.

If you're upgrading from {ServerlessProductName} 1.3.0 the operator will create a `KnativeServing` CR in the new API group for you. This CR will be a mirror of the `KnativeServing` CR in the old group, that was used in 1.3.0.
If you need to stick to the old group, you can use the old CR as before. Note though that it's deprecated and it will eventually be removed.
To upgrade to the new group, make sure all scripts and applications that rely on the old API group are updated to the new group. Once that is done, the old CR can be removed and the new one used for management.

Note: While the old CR is still there, changes to the new CR will be reconciled away by the operator. While the old CR is still active, all changes need to be made to it. Once you removed it, you can use the new CR as you used the old one in the past.
```

I'm sure this sounds awful, but I hope it gets across what we need to document here and what the users need to know 😂 

Essentially the migration path now looks like this:

1. Install the new version of the operators.
2.a. An installed system keeps working as before. You will have both an old and a new resource.
2.b. You can install the system freshly with the old resource and it still works as before. You will have both an old and a new resource.
2.c. You can install the system with the new CR. You will have only that new resource.
3. If you had both old and a new resource, you now have time to adjust your scripts etc. Once you've done that, you can remove the old resource without repercussions and continue to use just the new resource.

Including the note on that changes to the new CR will be undone while the old CR has not been deleted.

Reply:
Questions:
- why would someone need to stick to the old CR? It seems redundant to say this rather than just have everyone upgrade properly unless there's a reason.

I've made some updates based on not including mention of using the old CR for now. If there's a good use case for adding this lmk!

Reply:
I wanted to signify that the old CR still works and one essentially can use the lifespan of this entire version to do the switchover. That allows for graceful upgrades for consuming systems that depend on the old CR. You can‘t upgrade both versions at once without a conflict so you need to upgrade Serverless first and then until you upgrade the consuming system rely on the old CR to fully work.